src/OpenRiaServices.Hosting.AspNetCore/Framework/AspNetCore/OpenRiaServicesServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using Microsoft.Extensions.DependencyInjection;
+using OpenRiaServices.Server;
+
+namespace OpenRiaServices.Hosting.AspNetCore
+{
+    /// <summary>
+    /// Extension methods for registering OpenRiaServices services in an <see cref="IServiceCollection"/>
+    /// </summary>
+    public static class OpenRiaServicesServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds services required for using OpenRiaServices
+        /// </summary>
+        public static OpenRiaServicesOptionsBuilder AddOpenRiaServices(this IServiceCollection services)
+        {
+            ArgumentNullException.ThrowIfNull(services);
+
+            services.AddSingleton<OpenRiaServicesEndpointDataSource>();
+            services.AddSingleton(services);
```

Comment:
maybe we should just new upp OpenRiaServicesEndpointDataSource and pass in services in order to not register IServiceCollection

Reply:
could not do it easily since we requires access to the options instance